python/dask_cudf/dask_cudf/tests/test_dataframe.py

```diff
@@ -0,0 +1,73 @@
+import pandas as pd
+
+import cudf
+import dask.dataframe as dd
+import dask_cudf
+
+import pytest
+
+
+def test_get_dummies(data):
+    df = pd.DataFrame({"A": ["a", "b", "c", "a", "z"], "C": [1, 2, 3, 4, 5]})
+    df["B"] = df["A"].astype('category')
+    df["A"] = df["A"].astype('category')
+    ddf = dd.from_pandas(df, npartitions=10)
+    dd.assert_eq(dd.get_dummies(ddf).compute(), pd.get_dummies(df))
+    gdf = cudf.from_pandas(df)
+    gddf = dask_cudf.from_cudf(gdf, npartitions=10)
+    dd.assert_eq(dd.get_dummies(ddf).compute(), dd.get_dummies(gddf),
+                 check_dtype=False)
+
+    df = pd.DataFrame({"A": ["a", "b", "c", "a", "z"],
+                       "C": pd.Series([1, 2, 3, 4, 5], dtype='category')})
+    df["B"] = df["A"].astype('category')
+    df["A"] = df["A"].astype('category')
+    ddf = dd.from_pandas(df, npartitions=10)
+    dd.assert_eq(dd.get_dummies(ddf).compute(), pd.get_dummies(df))
+    gdf = cudf.from_pandas(df)
+    gddf = dask_cudf.from_cudf(gdf, npartitions=10)
+    dd.assert_eq(dd.get_dummies(ddf).compute(), dd.get_dummies(gddf),
+                 check_dtype=False)
+
+    df = pd.DataFrame({"C": pd.Series([1, 2, 3, 4, 5])})
+    ddf = dd.from_pandas(df, npartitions=10)
+    dd.assert_eq(dd.get_dummies(ddf).compute(), pd.get_dummies(df))
+    with pytest.raises(NotImplementedError):
+        dd.get_dummies(ddf, columns=['C']).compute()
+    gdf = cudf.from_pandas(df)
+    gddf = dask_cudf.from_cudf(gdf, npartitions=10)
+    with pytest.raises(NotImplementedError):
+        dd.get_dummies(gddf, columns=['C']).compute()
+    dd.assert_eq(dd.get_dummies(ddf).compute(), dd.get_dummies(gddf),
+                 check_dtype=False)
+
+    df = pd.DataFrame({"C": pd.CategoricalIndex([1, 2, 3, 4, 5])})
+    ddf = dd.from_pandas(df, npartitions=10)
+    dd.assert_eq(dd.get_dummies(ddf).compute(), pd.get_dummies(df))
+    gdf = cudf.from_pandas(df)
+    gddf = dask_cudf.from_cudf(gdf, npartitions=10)
+    dd.assert_eq(dd.get_dummies(ddf).compute(), dd.get_dummies(gddf),
+                 check_dtype=False)
+
+    df = pd.DataFrame({"C": [], "A": []})
+    ddf = dd.from_pandas(df, npartitions=10)
+    dd.assert_eq(dd.get_dummies(ddf).compute(), pd.get_dummies(df))
+    gdf = cudf.from_pandas(df)
+    gddf = dask_cudf.from_cudf(gdf, npartitions=10)
+    dd.assert_eq(dd.get_dummies(ddf).compute(), dd.get_dummies(gddf),
+                 check_dtype=False)
+
+    gdf = cudf.datasets.randomdata(nrows=200000, dtypes={"C": int,
```

Comment:
Given this is a unit test where we have concerns about runtime, can we make sure we don't introduce long-running tests? How long does this section of the test currently take to run?

Reply:
+1. If the concern is that a small data size may not result in enough partitions, then we need to figure out:

* the smallest number of partitions required to fully test correctness
* the smallest number of elements per partition required to fully test correctness 

- and figure out how to force the creation of such a dataset. Unit tests must be as small and as fast as possible.

Reply:
`test_get_dummies_large` (https://github.com/rapidsai/cudf/pull/2235/files#diff-bf8370b82b3eaa5737f32322702ad78eR61) takes 10 seconds to run.

Reply:
So given there's 100 partitions, what if we reduced the nrows down to say 2000 and got ~20 rows per partition? That way in theory the runtime drastically reduces and we still get our 100 partition test.

Reply:
Actually reducing number of partitions helps:

200000 records - 100 partitions - 45 seconds
**200000 records - 25 partitions - 13 seconds** <- current configuration
2000 records - 100 partitions - 45 seconds
2000 records - 25 partitions - 11 seconds
200 records - 25 partitions - 11 seconds

Reply:
One more set of results:
200000 records - 10 partitions - 8 seconds
20000 records - 10 partitions - 5 seconds <- I think this looks good ?? @beckernick @kkraus14

Reply:
5 seconds still sounds extremely slow, no? What's taking so long in the test?

Reply:
```python
In [10]: %time x = dd.get_dummies(gddf).compute()
CPU times: user 4.38 s, sys: 184 ms, total: 4.56 s
Wall time: 4.56 s

In [11]: %time x = dd.get_dummies(ddf).compute()
CPU times: user 60 ms, sys: 12 ms, total: 72 ms
Wall time: 68 ms
```

gddf being dask-cudf dataframe, ddf being dask dataframe.

Reply:
Yea, something feels extremely off here, any chance you can generate a snakeviz profile before we merge this?

Reply:
Sending this offline.

Reply:
Connected offline, there's performance concerns but they're somewhat unavoidable until we move more of this function down to the libcudf level.

@galipremsagar will open an issue to further discuss

Reply:
created an issue related to this: https://github.com/rapidsai/cudf/issues/2246